src/dfx/src/commands/generate.rs

```diff
@@ -22,7 +22,7 @@ pub struct GenerateOpts {
 }
 
 pub fn exec(env: &dyn Environment, opts: GenerateOpts) -> DfxResult {
-    let env = create_agent_environment(env, opts.network.network)?;
+    let env = create_anonymous_agent_environment(env, opts.network.network)?;
```

Comment:
Could this have been achieved like so?
```suggestion
    let env = create_agent_environment(env.with_identity_override(ANONYMOUS_IDENTITY_NAME), opts.network.network)?;
```